src/xcode/ENA/ENA/Source/Scenes/DynamicTableViewController/Views/DynamicTableViewRoundedCell.swift

```diff
@@ -38,39 +44,59 @@ class DynamicTableViewRoundedCell: UITableViewCell {
 		self.autoresizingMask = .flexibleHeight
 	}
 
-	private func setup(textStyle: ENAColor, backgroundStyle: ENAColor) {
+	private func setup(titleStyle: ENALabel.Style, bodyStyle: ENALabel.Style, textColor: ENAColor, bgColor: ENAColor, icons: [UIImage], buttonTitle: String?, buttonTapped: (() -> Void)?) {
 
 		// MARK: - General cell setup.
 		selectionStyle = .none
 		backgroundColor = .enaColor(for: .background)
 
 		// MARK: - Add inset view
-		insetView.backgroundColor = .enaColor(for: backgroundStyle)
+		insetView.backgroundColor = .enaColor(for: bgColor)
 		insetView.layer.cornerRadius = 16.0
 
 		// MARK: - Title adjustment.
-		title.style = .headline
-		title.textColor = .enaColor(for: textStyle)
+		title.style = titleStyle
+		title.textColor = .enaColor(for: textColor)
 		title.lineBreakMode = .byWordWrapping
 		title.numberOfLines = 0
 
 		// MARK: - Body adjustment.
-		body.style = .body
-		body.textColor = .enaColor(for: textStyle)
+		body.style = bodyStyle
+		body.textColor = .enaColor(for: textColor)
 		body.lineBreakMode = .byWordWrapping
 		body.numberOfLines = 0
 
+		// MARK: - Stackview adjustment.
+		if !icons.isEmpty {
+			iconStackView = UIStackView(frame: .zero)
```

Comment:
What do you mean by "Is this needed"? :) The setup method is called twice sometimes, and I want to make sure that there are not 4 icons instead of the 2 added ones :)

Reply:
Ah okay got it :D Thought this gets called only once and was wondering why it gets initialised again :)